src/runtime/graph/graph_runtime.cc

```diff
@@ -352,14 +376,14 @@ void GraphRuntime::SetupOpExecs() {
   for (uint32_t nid = 0; nid < this->GetNumOfNodes(); ++nid) {
     const auto& inode = nodes_[nid];
     if (inode.op_type == "null") continue;
-    std::vector<DLTensor> args;
+    std::vector<DLTensor*> args;
     for (const auto& e : inode.inputs) {
       uint32_t eid = this->entry_id(e);
-      args.push_back(*(data_entry_[eid].operator->()));
+      args.push_back(const_cast<DLTensor*>(data_entry_[eid].operator->()));
```

Comment:
Just to be clear, is this const_cast needed since the tensors are modified above on line 178-180?

```
  const_cast<DLTensor*>(old_t)->data = data_out->data;
  const_cast<DLTensor*>(old_t)->strides = data_out->strides;
  const_cast<DLTensor*>(old_t)->byte_offset = data_out->byte_offset;
```

Reply:
Hi Sam, sorry for getting back to you late. I forgot to reply after a five-day holiday. 

For the const_cast here, I think it would be necessary to keep it since the return type of operator->() is const DLTensor* and on line 423 in CreateTVMOp:

`arg_ptr->args = args;`

arg_ptr is a shared pointer to OpArgs and arg_ptr->args is std::vector<DLTensor*>. Elements of arg_ptr->args are later used for creating op_execs_. In case of breaking the runtime, it is probably better to do the const_cast here in stead of changing the following code.